hphp/runtime/vm/jit/vasm-simplify.cpp

```diff
@@ -524,6 +524,24 @@ bool simplify(Env& env, const pop& inst, Vlabel b, size_t i) {
 
 ///////////////////////////////////////////////////////////////////////////////
 
+bool simplify(Env& env, const movzbl& inst, Vlabel b, size_t i) {
+  // movzbl{s, d}; shrli{2, s, d} --> ubfm2r7ml{s, d}
+  return if_inst<Vinstr::shrli>(env, b, i + 1, [&](const shrli& sh) {
+    if (!(arch() == Arch::ARM &&
+      sh.s0.l() == 2 &&
+      env.use_counts[inst.d] == 1 &&
+      inst.d == sh.s1)) return false;
+
+  return simplify_impl(env, b, i, [&] (Vout& v) {
+    v << copy{inst.s, inst.d};
```

Comment:
struct movzbl { Vreg8 s; Vreg32 d; };
struct shrli { Immed s0; Vreg32 s1, d; VregSF sf; Vflags fl; };

The source in the move is Vreg8, the destination in the shrli is a Vreg32,
The copy uses Vreg.  Isn't that the correct way to match up the types?
Or do I not have to worry about that.